### ISSUE TITLE
Inlining bugs

### ISSUE THREAD

**Issue by lthls** [oxcaml/oxcaml#5438]
Consider the following program:
```ocaml
let[@inline never] f x = x

let[@inline always] g x = (f[@inlined always]) x

let h x = g x
```
Compiled in classic mode, we get a warning for the application of `f` inside `g` (as expected) but we also get a duplicate warning for the inlined version of `g` inside `h`. This wouldn't be too bad if the reported location mentioned the call to `f`, but it actually mentions the call to `g` instead so the error message doesn't make sense:
```text
File "/tmp/inlining_messages.ml", line 3, characters 26-48:
3 | let[@inline always] g x = (f[@inlined always]) x
                              ^^^^^^^^^^^^^^^^^^^^^^
Warning 55 [inlining-impossible]: Cannot inline: 
  the optimizer did not know what function was being applied (is it marked [@inline never]?)
  (the full inlining stack was: /tmp/inlining_messages.ml:3,26--48)

File "/tmp/inlining_messages.ml", line 5, characters 10-13:
5 | let h x = g x
              ^^^
Warning 55 [inlining-impossible]: Cannot inline: 
  the optimizer did not know what function was being applied (is it marked [@inline never]?)
  (the full inlining stack was: /tmp/inlining_messages.ml:5,10--13;/tmp/inlining_messages.ml:3,26--48)
```

It gets more interesting *without* classic mode: no warnings are emitted at all, because the call to `f` is actually inlined, ignoring the attribute on `f`. Looking at the inlining report, it seems that `Lambda_to_flambda` still performs inlining when not in classic mode, but without any check on the function attributes.
